tests/spec/fs.writeFile-readFile.spec.js

```diff
@@ -99,3 +99,132 @@ describe('fs.writeFile, fs.readFile', function() {
+    var fsPromises = util.fs().promises;
+
+    fsPromises.readFile('/no-such-file', 'utf8')
+      .then(data => expect(data).not.to.exist)
+      .catch(error => { 
+        expect(error).to.exist; 
+        expect(error.code).to.equal('ENOENT'); 
+        done(); 
+      });
+  });
+
+  it('should write, read a utf8 file without specifying utf8 in writeFile', function(done) {
+    var fsPromises = util.fs().promises;
+    var contents = 'This is a file.';
+
+    fsPromises.writeFile('/myfile', contents)
+      .then( () => {
+
+        fsPromises.readFile('/myfile', 'utf8')
+          .then(data => { 
+            expect(data).to.equal(contents);
+            done(); 
+          })
+          .catch(error => expect(error).not.to.exist);
+      })
+      .catch(error => { throw error; });
+  });
+
+  it('should write, read a utf8 file with "utf8" option to writeFile', function(done) {
+    var fsPromises = util.fs().promises;
+    var contents = 'This is a file.';
+
+    fsPromises.writeFile('/myfile', contents, 'utf8') 
+      .then( () => {
+
+        fsPromises.readFile('/myfile', 'utf8')
+          .then(data => { 
+            expect(data).to.equal(contents); 
+            done(); 
+          })
+          .catch(error => expect(error).not.to.exist);
+      })  
+      .catch(error => { throw error; });
+  });
+
+  it('should write, read a utf8 file with {encoding: "utf8"} option to writeFile', function(done) {
+    var fsPromises = util.fs().promises;
+    var contents = 'This is a file.';
+
+    fsPromises.writeFile('/myfile', contents, { encoding: 'utf8' })
+      .then( () => {
+
+        fsPromises.readFile('/myfile', 'utf8')
+          .then(data => { 
+            expect(data).to.equal(contents);
+            done(); 
+          })
+          .catch(error => expect(error).not.to.exist);
+      }) 
+      .catch(error => { throw error; });
+  });
+
+  it('should write, read a binary file', function(done) {
+    var fsPromises = util.fs().promises;
+    // String and utf8 binary encoded versions of the same thing: 'This is a file.'
+    var binary = new Buffer([84, 104, 105, 115, 32, 105, 115, 32, 97, 32, 102, 105, 108, 101, 46]);
+
+    fsPromises.writeFile('/myfile', binary)
+      .then( () => {
+
+        fsPromises.readFile('/myfile')
+          .then(data => { 
+            expect(data).to.deep.equal(binary); 
+            done(); 
+          })
+          .catch(error => expect(error).not.to.exist);
+      }) 
+      .catch(error => { throw error; });
+  });
+
+  it('should follow symbolic links', function(done) {
+    var fsPromises = util.fs().promises;
+    var contents = 'This is a file.';
+
+    fsPromises.writeFile('/myfile', '', { encoding: 'utf8' })
+      .then( () => {
+        fsPromises.symlink('/myfile', '/myFileLink')
+          .then( () => {
+            fsPromises.writeFile('/myFileLink', contents, 'utf8')
+              .then( () => {
+
+                fsPromises.readFile('/myFileLink', 'utf8')
```

Comment:
Same as before, move this function call inside the `then()` handler.
eg. `fsPromises.readFile('/myFileLink', 'utf8')`